cgpsa

```diff
@@ -1048,48 +1055,81 @@ sub checkMessageAddHeader
       {
         $headerlist = $sa_headers;
       }
-      
+
       foreach my $header (keys %{$headerlist})
       {
-        $newheaders .= 
+        $newheaders .=
           "X-Spam-".$header.": ".
           $rewritten_mail->get_pristine_header("X-Spam-".$header).
           "\e";
       }
-        
+
       if (!(exists $headerlist->{"Checker-Version"}))
       {
         # add X-Spam-Checker-Version if it isn't there already
 
         $newheaders .= "X-Spam-Checker-Version: SpamAssassin ".
                        Mail::SpamAssassin::Version().
                        " ($Mail::SpamAssassin::SUB_VERSION)\e";
-      }  
-        
+      }
+
       # always add CGPSA Version header and loop prevention header
-      
+
       $newheaders .= mandatory_headers("Scanned");
-      
-      # CommuniGate Pro can only take a limited-size response from the 
-      # plug-in (4K, in the versions of CommuniGate Pro released to date); 
+
+      # always add X-Junk-Score, if configured
+
+      if ($root_conf->{x_junk_score_header})
+      {
+        my $header_text = x_junk_score_header($status, $root_conf, "\e");
+
+        # if we can add the header, let's do it
+
+        if ((cgp_string_length($newheaders) +
+              cgp_string_length($header_text)) <
+            $root_conf->{max_response_length})
```

Comment:
This should be `$root_conf->{cgp_max_response_length})`

Reply:
Good catch, thanks! (this is what I get for not testing it myself)

Reply:
I believe I have seen instances of this mistake in the code base...

Reply:
I'll give it a scan.

Reply:
No, you did not, as far as I can tell. There are other settings (`max_scan_length`, `max_msg_length`) that don't have the `cgp` prefix.

Reply:
How about this line:
[cgpsa:1166@main](https://github.com/TFF-Enterprises/CGPSA/blob/434a2360e105a30fc4cd927e0b983b3e5b60a3a9/cgpsa#L1166)

Reply:
Already fixed in the version here (it's now line 1188).

Reply:
So I guess you did see it originally, but I'd already caught it back when you pointed out the other one.